packages/pluggableWidgets/progress-circle-web/src/components/__tests__/ProgressCircle.spec.tsx

```diff
@@ -0,0 +1,198 @@
+        expect(mockedAnimate).toHaveBeenCalledWith(0.23);
+    });
+
+    it("triggers an event when a clickable progress bar is clicked", () => {
+        const progressCircle = mount(
+            <ProgressCircle currentValue={23} minValue={0} maxValue={100} onClick={onClickSpy} label="23%" class="" />
+        );
+        progressCircle.find(".progress-circle-label-container").simulate("click");
+        expect(onClickSpy).toHaveBeenCalled();
+    });
+
+    it("handles a different range", () => {
+        const progressCircle = mount(
+            <ProgressCircle currentValue={40} minValue={20} maxValue={100} onClick={undefined} label="25%" class="" />
+        );
+        // Value 40 on range 20 - 100 is 25%.
+        expect(mockedAnimate).toHaveBeenCalledWith(0.25);
+        expect(progressCircle.text()).toBe("25%");
+    });
+
+    it("clamps a current value lower than the minimum value to 0% progress", () => {
+        const progressCircle = mount(
+            <ProgressCircle currentValue={-20} minValue={20} maxValue={100} onClick={undefined} label="0%" class="" />
+        );
+        expect(mockedAnimate).toHaveBeenCalledWith(0);
+        expect(progressCircle.text()).toContain("0%");
+    });
+
+    it("clamps a current value higher than the maximum value to 100% progress", () => {
+        const progressCircle = mount(
+            <ProgressCircle currentValue={102} minValue={20} maxValue={100} onClick={undefined} label="100%" class="" />
+        );
+        expect(mockedAnimate).toHaveBeenCalledWith(1);
+        expect(progressCircle.text()).toContain("100%");
+    });
+
+    it("is not clickable when there is no onClick handler provided", () => {
+        const progressCircle = mount(
+            <ProgressCircle
+                currentValue={-1}
+                minValue={0}
+                maxValue={100}
+                onClick={undefined}
+                label={undefined}
+                class=""
+            />
+        );
+        expect(
+            progressCircle.find(".progress-circle-label-container").hasClass("widget-progress-circle-clickable")
+        ).toBe(false);
+    });
+
+    describe("shows a runtime error Alert", () => {
+        it("when the current value is lower than the minimum value", () => {
+            const progressCircle = mount(
+                <ProgressCircle
+                    currentValue={-1}
+                    minValue={0}
+                    maxValue={100}
+                    onClick={onClickSpy}
+                    label={undefined}
+                    class=""
+                />
+            );
+            const alert = progressCircle.find(Alert);
+            expect(alert).toHaveLength(1);
+            expect(alert.text()).toBe(
+                "Error in progress circle values: The progress value is lower than the minimum value."
+            );
+        });
+        it("when the current value is higher than the maximum value", () => {
```

Comment:
Nit: new line - do we not have linting for this?

Reply:
don't think so